build_tools/python_deploy/build_linux_packages.sh

```diff
@@ -39,9 +39,31 @@
 # to packaging to avoid stomping on development artifacts.
 set -xeu -o errtrace
 
+# Function to find the directory the ".git" directory is in
+function find_git_dir_parent() {
+  # Get the current directory
+  curr_dir=$(pwd)
+
+  # Loop until we reach the root directory
+  while [ "$curr_dir" != "/" ]; do
+    # Check if there is a ".git" directory in the current directory
+    if [ -d "$curr_dir/.git" ]; then
+      # Return the path to the directory containing the ".git" directory
+      echo "$curr_dir"
+      return
+    fi
+
+    # Move up one directory
+    curr_dir=$(dirname "$curr_dir")
+  done
+
+  # If we reach the root directory and there is no ".git" directory, return an empty string
+  echo ""
+}
+
```

Comment:
```suggestion
# Function to find the directory the ".git" directory is in.
# We do this instead of using git directly because `git` may complain about
# operating in a directory owned by another user.
function find_git_dir_parent() {
  curr_dir="${PWD}"

  # Loop until we reach the root directory
  while [ "${curr_dir}" != "/" ]; do
    # Check if there is a ".git" directory in the current directory
    if [ -d "${curr_dir}/.git" ]; then
      # Return the path to the directory containing the ".git" directory
      echo "${curr_dir}"
      return
    fi

    # Move up one directory
    curr_dir="$(dirname "${curr_dir}")"
  done

  # If we reach the root directory and there is no ".git" directory, return an empty string
  echo ""
}

```